Add Hardware Wallet integration guidance (#4108)
zkSNACKs/WalletWasabi: WalletWasabi.Documentation/Guides/HardwareWalletIntegration.md

Wasabi's main priorities regarding hardware wallets:
- Compatibility
- Privacy

> 

Reviewer: ```suggestion
- Industry standardization

```

> 1. Integrate your hardware wallet into HWI https://github.com/bitcoin-core/HWI

Reviewer 2: ```suggestion
1. Integrate your hardware wallet into [HWI](https://github.com/bitcoin-core/HWI).
```

> 2. Add some tests into HWI, so compatibility problems can be caught at early stages

Reviewer 3: ```suggestion
2. Add some tests into HWI, so compatibility issues can be caught at early stages.
```

> 3. Test the compatibility with Wasabi. Import, Receive, Send, Recover.

Reviewer 3: ```suggestion
3. Test the compatibility with Wasabi. Import, Receive, Send and Recover.
```

> 4. Write some Kata tests (manual tests), that can be used to detect compatibility issues when: Wasabi release, new firmware, etc. https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Tests/AcceptanceTests/HwiKatas.cs

Reviewer 3: What is Kata test? Is that a Wasabi thing? Is that some general way of testing?

Reviewer 2: https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Tests/AcceptanceTests/HwiKatas.cs#L16-L22

> 4. Write some Kata tests (manual tests), that can be used to detect compatibility issues when: Wasabi release, new firmware, etc. https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Tests/AcceptanceTests/HwiKatas.cs

Reviewer 2: ```suggestion
4. Write some [Kata tests](https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Tests/AcceptanceTests/HwiKatas.cs) (manual tests), that can be used to detect compatibility issues when: Wasabi release, new firmware, etc.
```

> 5. Send at least 1 device (preferable two devices) to Wasabi HQ, Contact us at info@zksnacks.com

Reviewer 3: ```suggestion
5. Send at least 1 device (preferable two devices) to Wasabi HQ. To do so, contact us at <info@zksnacks.com>.
```

Reviewer 2: preferable -> preferably

6. Wasabi team tests the device using the Kata tests and approves.

> 7. Create some content on how to use the device with our software.

Reviewer: ```suggestion
7. Create some content on how to use the device with our software, possibly in [the Wasabi documentation](https://github.com/zkSNACKs/WasabiDoc/blob/master/docs/using-wasabi/ColdWasabi.md).
```

> 8. Create a guide on how to initialize the device in offline mode - if possible. There should be a method to init the device without sharing the XPUB.

Reviewer: ```suggestion
8. Create a guide on how to initialize the device in offline mode - if possible. There should be a method to init the device without sharing the xpub.
```

> 9. Wasabi unofficially supports the hardware wallet - meaning that it is working but nothing is guaranteed. 

Reviewer: I'm not sure about this...

Right now, Wasabi officially supports ledger and coldcard devices. Will these be unofficially supported at some point too? Under which criteria?

When a new device gets added, it is at first unofficially supported. But when does such a device move on to officially supported?

But this is a more general project design decision, so out of scope for this PR, I guess. 

However, this sentence is still confusing, I would rewrite it accordingly.

> 10. Half-year grace period - without compatibility problems or breaking changes

Reviewer: ```suggestion
10. After a half-year grace period without compatibility problems or breaking changes, Wasabi will officially supports this hardware wallet, meaning that Wasabi will give user guidance related to the device and announce the [support of the device](https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Documentation/WasabiCompatibility.md).
```

Reviewer 2: supports -> support

> 11. Wasabi officially supports the hardware wallet - meaning that Wasabi will give support related to the device and announce the support of the device. https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Documentation/WasabiCompatibility.md 

Reviewer: ```suggestion
```

# Hardware wallet integration into Wasabi 

## Introduction 

This is a technical document written for Hardware Wallet manufacturers. It describes the steps that need to be done before a hardware wallet can be supported by Wasabi. 

> Wasabi not directly supports Hardware Wallets but it uses [HWI](https://github.com/bitcoin-core/HWI) which is a command-line tool that unifies the commands for many devices. Wasabi includes the HWI binary and calling it every time whenever interacts with the device. 

Reviewer: ```suggestion
Wasabi does not directly support Hardware Wallets, but it uses [Bitcoin Core's Hardware Wallet Interface (HWI)](https://github.com/bitcoin-core/HWI) which is a command-line tool that unifies the commands for many devices. Wasabi includes the HWI binary and calls it every time whenever it interacts with the device. 
```

Wasabi's main priorities regarding hardware wallets:
- Compatibility - past, present, future
- Privacy
- Industry standardization

## Integration procedure

1. Integrate your hardware wallet into [HWI](https://github.com/bitcoin-core/HWI).
2. Add some tests into HWI, so compatibility issues can be caught at the early stages.
3. Test the compatibility with Wasabi. Import, Receive, Send, and Recover.
4. Write some [Kata tests](https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi.Tests/AcceptanceTests/HwiKatas.cs) (manual tests), that can be used to detect compatibility issues when: Wasabi release, new firmware, etc.

> 5. Send at least 1 device (preferably two devices) to Wasabi HQ. To do so, contact us at `info@zksnacks.com`.

Reviewer: ```suggestion
5. Send at least one device (preferably two devices) to Wasabi HQ. To do so, contact us at `info@zksnacks.com`.
```
